### PR TITLE
Return empty list of indexes if collection doesn't exist

### PR DESCRIPTION
**Pull request by konradwudkowski**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://github.com/ReactiveMongo/ReactiveMongo/blob/master/CONTRIBUTING.md#reactivemongo-developer--contributor-guidelines)?
* [x] Have you added tests for any changed functionality?

## Purpose

Return empty list when listing indexes on a collection that doesn't exist yet.

### CODE REVIEW

File: driver/src/main/scala/api/indexes/IndexesManager.scala [hunk 1]
```diff
@@ -271,10 +271,14 @@ private class DefaultCollectionIndexesManager(db: DB, collectionName: String)(
...
       collection, listCommand, ReadPreference.primary).recoverWith {
-      case err: WriteResult if err.code.exists(_ == 26 /* no database */ ) =>
+      case err: WriteResult if err.code.contains(26 /* no databse */ ) =>
+        Future.successful(List.empty[Index])
+
+      case err: DefaultBSONCommandError if err.code.contains(26 /* no collection */ ) =>
         Future.successful(List.empty[Index])
 
-      case err => Future.failed(err)
+      case err =>
```
[the remark this answers is not in the document]
Comment (konradwudkowski) [reply driver/src/main/scala/api/indexes/IndexesManager.scala]: kept `Future.failed(err)` on the same line

File: driver/src/test/scala/IndexesSpec.scala [hunk 2]
```diff
@@ -185,6 +185,14 @@ class IndexesSpec(implicit ee: ExecutionEnv)
     }
   }
 
+  "Listing indexes" should {
+    "return empty list if collection doesn't exist" in {
+      lazy val col = db(s"nonexistent-collection-$hashCode")
+
+      col.indexesManager.list() must beEqualTo(Nil).await(0, timeout)
```
[the remark this answers is not in the document]
Comment (konradwudkowski) [reply driver/src/test/scala/IndexesSpec.scala]: changed to beEmpty

File: driver/src/main/scala/api/indexes/IndexesManager.scala [hunk 3]
```diff
@@ -271,10 +271,14 @@ private class DefaultCollectionIndexesManager(db: DB, collectionName: String)(
   def list(): Future[List[Index]] =
     runner(
       collection, listCommand, ReadPreference.primary).recoverWith {
-      case err: WriteResult if err.code.exists(_ == 26 /* no database */ ) =>
+      case err: WriteResult if err.code.contains(26 /* no databse */ ) =>
```
[the remark this answers is not in the document]
Comment (konradwudkowski) [reply driver/src/main/scala/api/indexes/IndexesManager.scala]: `DefaultBSONCommandError` is not a `WriteResult` and they don't have a common ancestor with `error.code` to simplify it

**Comment by konradwudkowski** [comment]
@cchantep I've made changes following your suggestions, however in one case I didn't know how ("case WriteResult.code(26) must be able to catch both theses case") and I explained why after your comment.

File: driver/src/main/scala/api/indexes/IndexesManager.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (cchantep) [reply driver/src/main/scala/api/indexes/IndexesManager.scala]: http://reactivemongo.org/releases/0.1x/api/reactivemongo/api/commands/CommandError$$Code$.html#unapply(scrutinee:Any):Option[Int]

**Comment by konradwudkowski** [comment]
@cchantep I've used single `case CommandError.Code` to match both cases

**Pull request merged**